### PR TITLE
[Objectarium] ✔ Add more input validation

### PR DESCRIPTION
**Pull request by bdeneux**
According to #560 

Given https://github.com/axone-protocol/contracts/issues/560#issuecomment-2147158351 about `max_object_pin`  and uncertainty surrounding potential use cases, I believe it's best to avoid imposing excessive constraints and ensures flexibility for future use cases that may arise 🤷.

## Summary by CodeRabbit

- **New Features**
  - Enhanced validation logic for bucket limits and pagination settings, ensuring more robust initialization.
  
- **Bug Fixes**
  - Improved error handling during contract instantiation, reducing potential issues with configuration and limits.

- **Documentation**
  - Clarified the description of the `compression_algorithms` field in the bucket configuration, improving user understanding of acceptable values.

### CODE REVIEW

File: contracts/axone-objectarium/src/contract.rs [hunk 1]
```diff
@@ -582,33 +582,76 @@ mod tests {
...
+            ),
+            (
+                BucketLimitsBuilder::default()
+                    .max_total_size(10u128)
+                    .max_object_size(20u128)
+                    .build()
+                    .unwrap(),
+                Default::default(),
+                StdError::generic_err("'max_total_size' cannot be less than 'max_object_size'"),
+            ),
```
[the remark this answers is not in the document]
Comment (bdeneux) [reply contracts/axone-objectarium/src/contract.rs:643]: @coderabbitai, thank you for your suggestion. However, I don't believe it brings much clarity compared to the previous version, especially considering that other test cases work by directly setting the concerned errors in the cases Vec, not just the string errors.

File: contracts/axone-objectarium/src/state.rs [hunk 2]
```diff
@@ -194,14 +194,58 @@ impl From<msg::BucketLimits> for BucketLimits {
...
+    fn try_new(
+        max_total_size: Option<Uint128>,
+        max_objects: Option<Uint128>,
+        max_object_size: Option<Uint128>,
+        max_object_pins: Option<Uint128>,
+    ) -> StdResult<BucketLimits> {
+        if let Some(max_total_size) = max_total_size {
+            if max_total_size == Uint128::zero() {
+                return Err(StdError::generic_err("'max_total_size' cannot be zero"));
+            }
```
Comment (ccamel) [inline contracts/axone-objectarium/src/state.rs:207]: I always feel like using the [`ensure!`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/macro.ensure.html) macro in such cases as it conveys the intent of a precondition that must be met.

In the specific case of checking if the `max_total_size` is not 0,  we can streamline the precondition check to a single line using the  [`ensure_ne`](https://docs.rs/cosmwasm-std/latest/cosmwasm_std/macro.ensure_ne.html) macro:

```rust
ensure_ne!(max_total_size, Some(Uint128::zero()), StdError::generic_err("'max_total_size' cannot be zero"));
```

Comment (bdeneux) [reply contracts/axone-objectarium/src/state.rs:207]: Oh yes you're right, I have a short memory 😅

**Review by ccamel — APPROVED** [review]
Great 👍

**Review by amimart — CHANGES_REQUESTED** [review]
Nice!

I think as mentioned in the issue we also need to ensure that `accepted_compression_algorithms`, even if there is a serde default, when specified in the instantiate message as empty array it is accepted for the moment, the actual documentation is not aligned with the behaviour.

**Comment by bdeneux** [comment]
> Nice!
> 
> I think as mentioned in the issue we also need to ensure that `accepted_compression_algorithms`, even if there is a serde default, when specified in the instantiate message as empty array it is accepted for the moment, the actual documentation is not aligned with the behaviour.


@amimart  Yes ! I will update the documentation according to this behavior and add the corresponding tests and implementation.

**Review by amimart — APPROVED** [review]
Looks good to me, thanks! 🙏

**Pull request merged**